### PR TITLE
chore(jsdoc): add JSDoc to src/runtime/styles.ts

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
This adds explanations for what goes on in this file, since before things were a bit mysterious.

## What is the current behavior?

GitHub Issue Number: N/A

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by christian-bromann — APPROVED** [review]
💯

File: src/runtime/styles.ts [hunk 1]
```diff
@@ -8,10 +8,21 @@ import { HYDRATED_STYLE_ID, NODE_TYPE, SLOT_FB_CSS } from './runtime-constants';
...
+ *
+ * @param scopeId the scope ID for the component of interest
+ * @param cssText styles for the component of interest
+ * @param allowCS whether or not to use a constructable stylesheet
+ */
 export const registerStyle = (scopeId: string, cssText: string, allowCS: boolean) => {
   let style = styles.get(scopeId);
   if (supportsConstructableStylesheets && allowCS) {
-    style = (style || new CSSStyleSheet()) as CSSStyleSheet;
+    style ||= new CSSStyleSheet();
```
Comment (rwaskiewicz) [inline src/runtime/styles.ts:25]: Does this get transpiled down in any way in the build output? Technically, we still support Chrome/Edge 79-84 (inclusive) that don't support logical-or assignment - https://caniuse.com/?search=logical%20assignment

File: src/runtime/styles.ts [hunk 2]
```diff
@@ -23,6 +34,19 @@ export const registerStyle = (scopeId: string, cssText: string, allowCS: boolean
   styles.set(scopeId, style);
 };
 
+/**
+ * Attach the styles for a given component to the DOM
+ *
+ * If the element uses shadow or is already attached to the DOM then we can
+ * create a stylesheet inside of it's associated document fragment, otherwise
```
Comment (rwaskiewicz) [inline src/runtime/styles.ts:41]: Non-blocking nit
```suggestion
 * create a stylesheet inside of its associated document fragment, otherwise
```

**Review by rwaskiewicz — APPROVED** [review]

File: src/runtime/styles.ts (hunk 1 above)
Comment (alicewriteswrongs) [reply src/runtime/styles.ts:25]: ah I thought we were good on that now, I'll just get rid of that

**Pull request merged**